teps/0023-implicit-mapping.md

```diff
@@ -0,0 +1,358 @@
+---
+title: 0023-Implicit-parameter-mapping
+authors:
+  - "@Peaorl"
+  - "@wlynch"
+creation-date: 2020-10-01
+last-updated: 2021-05-20
+status: proposed
+---
+
+# TEP-0023: Implicit Parameter Mapping for Embedded Specs
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goal](#goal)
+  - [Non Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Caveats](#caveats)
+    - [Naming conflicts](#common-parameter-and-resource-names)
+    - [Common parameter names](#common-parameter-names)
+  - [Performance](#performance)
+- [Design Details](#design-details)
+  - [Alternatives](#alternatives)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Future Work](#future-work)
+- [References](#references)
+
+## Summary
+
+Pipeline specs can sometimes feel unnecessarily verbose, particularly when the
+same fields need to be passed around different Tasks again and again.
+
+In this TEP, we propose introducing implicit parameter mapping to allow authors
+to reference params in embedded specs that are not explicitly defined. Instead,
+users will rely on Tekton to automatically resolve these values at admission
+time.
+
+## Motivation
+
+We want to make it easier for users to get started with Tekton by reducing some
+of the complexity in writing specs. One way we wish to do this is to reduce the
+amount of repetition in Pipeline specs, inferring information as much as
+possible. Our hope is that this will make it easier to get started with simple
+PipelineRun definitions and let users focus on the core functionality of their
+Pipelines.
```

Comment:
wow! this anology very relevant in explaining Tekton Concepts.
Instead of explaining Tasks and then TaskRuns, showing a TaskRun being run with a TaskSpec first and then telling that the TaskSpec part can be moved to a parameterized reusable Task, like a function in programming, immediately sticks. 🧑‍💻 🧑‍🎓 

credit @sthaha  🙇